pyartifactory/objects.py

```diff
@@ -784,17 +812,24 @@ def deploy(
         :param artifact_path: Path to file in Artifactory
         :param local_file_location: Location of the file to deploy
         """
-        artifact_path = artifact_path.lstrip("/")
-        local_filename = artifact_path.split("/")[-1]
-        with open(local_file_location, "rb") as file:
-            self._put(f"{artifact_path}", data=file)
-            logger.debug("Artifact %s successfully deployed", local_filename)
-            return self.info(artifact_path)
+        if isdir(local_file_location):
+            for root, _, files in os.walk(local_file_location):
+                new_root = f"{artifact_path}/{root[len(local_file_location):]}"
+                for file in files:
+                    self.deploy(join(root, file), f"{new_root}/{file}")
+        else:
+            artifact_path = artifact_path.lstrip("/")
+            local_filename = artifact_path.split("/")[-1]
+            with open(local_file_location, "rb") as file:  # type: ignore[assignment]
+                self._put(f"{artifact_path}", data=file)
```

Comment:
I think I would rather change the variable name from `file` to something else than ignore a mypy error. Here mypy is confused because there is already a `file` variable declared above (in the `for` loop) with the type `str`, so it doesn't want you to reassign it to a `BinaryIO` type.
Something like this fixes it (but you can choose another name if you want):
```suggestion
            with open(local_file_location, "rb") as file_to_upload:
                self._put(f"{artifact_path}", data=file_to_upload)
```

Reply:
Here I actually disagree: there are actually two distinct variables (within non-overlapping scopes), so it's actually a shortcoming of the `mypy` tool not to support this -yet. (There are discussions to improve that, see https://github.com/python/mypy/issues/6232 for example.)

Reply:
Oh, I see. Too bad about mypy ^^'
I don't know how to handle it, there is a workaround to avoid bypassing the error but it's annoying. I'm afraid we will forget it there and silence too many errors without knowing it.

Reply:
I can add a comment to explain the issue. (But I do have a rather strong aversion against changing perfectly fine code just to make some tool happy ;-) )